components/tidb_query_expr/src/impl_cast.rs

```diff
@@ -943,7 +943,7 @@ fn cast_int_as_duration(
                     ctx.handle_truncate_err(err)?;
                     Ok(None)
                 } else {
-                    Err(err.into())
+                    Ok(ctx.handle_invalid_duration_error(err).map(|_| None)?)
```

Comment:
Let's add a comment here.

And only case int as duration affected?

Reply:
oh sorry, I think this PR is kind of duplicate with another one. I will close it. Thanks for your quick response.